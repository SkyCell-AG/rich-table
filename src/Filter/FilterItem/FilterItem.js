import React from 'react'
import PropTypes from 'prop-types'
import ListItem from '@material-ui/core/ListItem'
import Chip from '@material-ui/core/Chip'

import Checkbox from 'Checkbox'

import useStyles from './FilterItem.styles'

const propTypes = {
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    count: PropTypes.number.isRequired,
    checked: PropTypes.bool.isRequired,
    onClick: PropTypes.func.isRequired,
    disabled: PropTypes.bool.isRequired,
}

const FilterItem = (props) => {
    const {
        count,
        label,
        value,
        checked,
        disabled,
        onClick,
    } = props

    const classes = useStyles()

    return (
        <ListItem
            key={value}
            button
            disabled={disabled}
            className={classes.root}
            onClick={onClick}
        >
            <div className={classes.labelCheckbox}>
                <Checkbox
                    value={checked}
                    title={label === undefined ? value : label}
                    disabled={disabled}
                />
            </div>
            {count && (
                <Chip
                    label={count}
                    size="small"
                />
            )}
        </ListItem>
    )
}

FilterItem.propTypes = propTypes

export default FilterItem
