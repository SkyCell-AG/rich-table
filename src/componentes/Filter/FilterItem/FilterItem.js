import React from 'react'
import PropTypes from 'prop-types'
import ListItem from '@mui/material/ListItem'
import Chip from '@mui/material/Chip'

import Checkbox from 'componentes/Checkbox'

import useStyles from './FilterItem.styles'

const propTypes = {
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    count: PropTypes.number.isRequired,
    checked: PropTypes.bool.isRequired,
    onClick: PropTypes.func.isRequired,
    disabled: PropTypes.bool.isRequired,
}

const FilterItem = (props) => {
    const {
        count,
        label,
        value,
        checked,
        disabled,
        onClick,
    } = props

    const classes = useStyles()

    return (
        <ListItem
            key={value}
            button
            disabled={disabled}
            className={classes.root}
            onClick={onClick}
        >
            <div className={classes.labelCheckbox}>
                <Checkbox
                    value={checked}
                    title={label === undefined ? value : label}
                    disabled={disabled}
                />
            </div>
            {count && (
                <Chip
                    label={count}
                    size="small"
                />
            )}
        </ListItem>
    )
}

FilterItem.propTypes = propTypes

export default FilterItem
