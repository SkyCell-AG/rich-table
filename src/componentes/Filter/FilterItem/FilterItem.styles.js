import {
    makeStyles,
} from '@material-ui/core'

const useStyles = makeStyles(() => {
    return {
        root: {
            display: 'flex',
            justifyContent: 'space-between',
            paddingTop: 0,
            paddingBottom: 0,
            height: '40px !important',
        },
        labelCheckbox: {
            display: 'flex',
            justifyContent: 'flex-start',
            alignItems: 'center',
        },
    }
})

export default useStyles
