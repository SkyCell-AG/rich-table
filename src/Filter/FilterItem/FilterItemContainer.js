import React from 'react'

import FilterItem from './FilterItem'

const FilterItemContainer = (props) => {
    return (
        <FilterItem
            {...props}
        />
    )
}

export default FilterItemContainer
