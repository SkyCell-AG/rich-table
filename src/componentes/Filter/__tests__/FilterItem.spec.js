import React from 'react'
import {
    shallow,
} from 'enzyme'

import FilterItem from '../FilterItem'

describe('app/shared-components/Filter/FilterItem should match snapshot', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(
            <FilterItem
                count={10}
                label="2500C"
                value="9x1EF20BD4D474D284ED257A1DAE235766F6EB3C2255BB0901832616UY"
                checked
                disabled={false}
                onClick={jest.fn()}
            />,
        )

        expect(wrapper).toMatchSnapshot()
    })
})
