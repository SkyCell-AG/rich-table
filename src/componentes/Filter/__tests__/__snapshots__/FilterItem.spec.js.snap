// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/shared-components/Filter/FilterItem should match snapshot should match snapshot 1`] = `
<FilterItem
  checked={true}
  count={10}
  disabled={false}
  label="2500C"
  onClick={[MockFunction]}
  value="9x1EF20BD4D474D284ED257A1DAE235766F6EB3C2255BB0901832616UY"
/>
`;
